pkg/schemavalidator/schemavalidator_test.go

```diff
@@ -0,0 +1,342 @@
+			expected: false,
+		},
+		{
+			name:     "Valid URL Format",
+			profile:  `{"primary_url": "https://example.com"}`,
+			expected: true,
+		},
+		{
+			name:     "Invalid URL in Images Array",
+			profile:  `{"images": [{"name": "Test", "url": "ftp://invalid.com"}]}`,
+			expected: false,
+		},
+		{
+			name:     "Valid Image URL but Missing Required Field",
+			profile:  `{"images": [{"name": "Test"}]}`,
+			expected: false,
+		},
+		{
+			name:     "Valid Geolocation",
+			profile:  `{"geolocation": {"lat": 45.123, "lon": -75.123}}`,
+			expected: true,
+		},
+		{
+			name:     "Invalid Geolocation (Latitude Out of Bounds)",
+			profile:  `{"geolocation": {"lat": 95.123, "lon": -75.123}}`,
+			expected: false,
+		},
+		{
+			name:     "Invalid Linked Schemas Pattern",
+			profile:  `{"linked_schemas": ["InvalidSchema-v0.1.0"]}`,
+			expected: false,
+		},
+		{
+			name:     "Valid Linked Schemas Pattern",
+			profile:  `{"linked_schemas": ["validschema_12345-v0.1.0"]}`,
+			expected: true,
+		},
+		{
+			name:     "Invalid ISO 3166 Country Code",
+			profile:  `{"country_iso_3166": "USA"}`,
+			expected: false,
+		},
+		{
+			name:     "Valid ISO 3166 Country Code",
+			profile:  `{"country_iso_3166": "AE"}`,
+			expected: true,
+		},
+		// {
+		// 	name:     "Overly Long String for Description",
+		// 	profile:  `{"description": "` + strings.Repeat("a", 5000) + `"}`,
+		// 	expected: false,
+		// },
+		// {
+		// 	name:     "Invalid Email in Contact Details",
+		// 	profile:  `{"contact_details": {"email": "invalid.email", "contact_form": "https://valid.com"}}`,
+		// 	expected: false,
+		// },
+		// {
+		// 	name:     "Non-UTF8 Characters in Name",
+		// 	profile:  `{"name": "` + "\xC3\x28" + `"}`, // An invalid UTF-8 sequence
+		// 	expected: false,
+		// },
+		// {
+		// 	name:     "SQL Injection Attempt in Name",
+		// 	profile:  `{"name": "Robert'); DROP TABLE Students;--"}`, // A basic SQL injection payload
+		// 	expected: false,
+		// },
+		// {
+		// 	name:     "NoSQL Injection Attempt in Name",
+		// 	profile:  `{"name": "{$ne: null}"}`, // A basic MongoDB NoSQL injection payload
+		// 	expected: false,
+		// },
+		// {
+		// 	name:     "NoSQL Injection Attempt in Description",
+		// 	profile:  `{"description": "{$gt: ''}"}`, // Another MongoDB NoSQL injection payload
+		// 	expected: false,
+		// },
+		// {
+		// 	name:     "XSS Payload in Image Name",
+		// 	profile:  `{"images": [{"name": "<script>alert('xss')</script>", "url": "https://valid.com/image.jpg"}]}`, // A basic XSS payload
+		// 	expected: false,
+		// },
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			validator, err := schemavalidator.NewBuilder().
+				WithStrSchemas([]string{StrSchema}).
+				WithStrProfile(tt.profile).
+				Build()
+
+			require.NoError(t, err)
+			result := validator.Validate()
+			require.Equal(t, tt.expected, result.Valid)
+		})
+	}
+}
+
+// func FuzzValidate(f *testing.F) {
```

Comment:
Go has a built-in fuzzing test package: https://go.dev/security/fuzz. The problem is that the values generated are random. I'm not sure how we are going to validate its correctness.